multitenancy/src/main/java/org/axonframework/extensions/multitenancy/components/MultiTenantSubscribableMessageSource.java

```diff
@@ -0,0 +1,27 @@
+package org.axonframework.extensions.multitenancy.components;
+
+import java.util.Map;
+
+/**
+ * Interface for multi-tenant message sources that can provide tenant segments.
+ *
+ * @param <T> The type of the tenant segment
+ */
+import org.axonframework.eventhandling.EventMessage;
+import org.axonframework.messaging.SubscribableMessageSource;
+
+/**
+ * Interface for multi-tenant message sources that can provide tenant segments.
+ *
+ * @param <T> The type of the tenant segment, which must extend MessageSource
+ */
+public interface MultiTenantSubscribableMessageSource<T extends SubscribableMessageSource<EventMessage<?>>> {
+
+    /**
+     * Returns a map of tenant segments, where the key is the TenantDescriptor
+     * and the value is the corresponding tenant segment of type T, which extends MessageSource.
+     *
+     * @return A map of TenantDescriptor to tenant segments
```

Comment:
Text is right! Just a minor linking detail:
```suggestion
     * Returns a map of tenant segments, where the key is the {@link TenantDescriptor}
     * and the value is the corresponding tenant segment of type T, which extends {@link SubscribableMessageSource}.
     *
     * @return A map of {@link TenantDescriptor} to tenant segments
```